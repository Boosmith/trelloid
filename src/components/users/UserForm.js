import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";

const UserForm = ({ user, onChange, onSave, errors }) => {
	return (
		<form onSubmit={onSave}>
			<h1>Manage User</h1>
			<TextInput
				name="firstName"
				label="First name"
				value={user.firstName}
				onChange={onChange}
				error={errors.firstName}
			/>

			<TextInput
				name="lastName"
				label="Last name"
				value={user.lastName}
				onChange={onChange}
				error={errors.lastName}
			/>

			<input type="submit" value="Save" className="btn btn-primary" />
		</form>
	);
};

UserForm.propTypes = {
	user: PropTypes.object.isRequired,
	onChange: PropTypes.func.isRequired,
	onSave: PropTypes.func.isRequired,
	errors: PropTypes.object
};

export default UserForm;
