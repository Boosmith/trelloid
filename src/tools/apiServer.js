const jsonServer = require("json-server");
const server = jsonServer.create();
const path = require("path");
const fs = require("fs");
const jwt = require("jsonwebtoken");
const bcrypt = require("bcrypt");
const router = jsonServer.router(path.join(__dirname, "db.json"));
const users = JSON.parse(
	fs.readFileSync(path.join(__dirname, "db.json"), "UTF-8")
).users;
const ObjectId = require("bson").ObjectId;

const middlewares = jsonServer.defaults();

server.use(middlewares);

server.use(jsonServer.bodyParser);

server.use(function(req, res, next) {
	setTimeout(next, 2000);
});

server.use((req, res, next) => {
	if (req.method === "POST") {
		req.body.createdAt = Date.now();
	}
	next();
});

server.post("/users/", function(req, res, next) {
	const error = validateUser(req.body);
	if (error) {
		res.status(400).send(error);
	} else {
		req.body._id = createMongoId();
		next();
	}
});

server.post("/users/authenticate", function(req, res, next) {
	const { userName, password } = req.body;
	if (isAuthenticated(userName, password) === false) {
		const status = 401;
		const message = "Incorrect email or password";
		res.status(status).json({ status, message });
		return;
	}
	const access_token = createToken({ userName, password });
	res.status(200).json({ access_token });
});

router.db._.id = "_id";

server.use("/api", router);

const port = 3001;
server.listen(port, () => {
	console.log(`JSON Server is running on port ${port}`);
});

function createMongoId() {
	return new ObjectId().toString();
}

function validateUser(user) {
	if (!user.firstName) return "First name is required.";
	if (!user.lastName) return "Last name is required.";
	if (!user.address) return "Address is required.";
	if (!user.city) return "City is required.";
	if (!user.postcode) return "Postcode is required.";
	return "";
}

const SECRET_KEY = "123456789";
const expiresIn = "1h";

function createToken(payload) {
	return jwt.sign(payload, SECRET_KEY, { expiresIn });
}

function verifyToken(token) {
	return jwt.verify(token, SECRET_KEY, (err, decode) =>
		decode !== undefined ? decode : err
	);
}

function isAuthenticated(userName, password) {
	return (
		users.findIndex(
			user =>
				user.userName === userName &&
				bcrypt.compareSync(password, user.password)
		) !== -1
	);
}
