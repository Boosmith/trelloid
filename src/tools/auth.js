export function bearerHeaders(token) {
	return {
		headers: { Authorization: "Bearer " + token }
	};
}
