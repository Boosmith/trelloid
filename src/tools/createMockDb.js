/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const { boards, cards, cardStatuses, users } = mockData;
const data = JSON.stringify({ boards, cards, cardStatuses, users });
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function(err) {
	err ? console.log(err) : console.log("Mock DB created.");
});
