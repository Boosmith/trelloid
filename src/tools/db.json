{
  "users": [
    {
      "_id": "5d42e5d5e1e3d731df26010a",
      "firstName": "Andrew",
      "lastName": "Smith",
      "address": "32 Walpole Road",
      "city": "London",
      "postcode": "N17",
      "__v": 0
    },
    {
      "_id": "5d43fed51d0825849c0f0b67",
      "firstName": "Rosa",
      "lastName": "Smith",
      "address": "32 Walpole Road",
      "city": "London",
      "postcode": "N17",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129604",
      "firstName": "Dora",
      "lastName": "Fitzgerald",
      "address": "1300 Eknaw Path",
      "city": "Evegirhe",
      "postcode": "P8K 4T1",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129605",
      "firstName": "Bryan",
      "lastName": "Fitzgerald",
      "address": "1050 Dedu Point",
      "city": "Irhekde",
      "postcode": "G1K 2N9",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129606",
      "firstName": "Dale",
      "lastName": "Fitzgerald",
      "address": "560 Fibab Glen",
      "city": "Hudmakap",
      "postcode": "T9U 0N6",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129607",
      "firstName": "Alma",
      "lastName": "Fitzgerald",
      "address": "1630 Zuas Place",
      "city": "Caturaagu",
      "postcode": "G9E 7N0",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129608",
      "firstName": "Billy",
      "lastName": "Fitzgerald",
      "address": "1689 Hopge Square",
      "city": "Vinpati",
      "postcode": "C2O 6D1",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c129609",
      "firstName": "Adeline",
      "lastName": "Fitzgerald",
      "address": "1912 Dusved Ridge",
      "city": "Atehasma",
      "postcode": "N2D 7C3",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c12960a",
      "firstName": "Charlotte",
      "lastName": "Fitzgerald",
      "address": "611 Gumaf Way",
      "city": "Kalover",
      "postcode": "E4Y 2E7",
      "__v": 0
    },
    {
      "_id": "5d42104a7979bd071c12960b",
      "firstName": "Rodney",
      "lastName": "Fitzgerald",
      "address": "1961 Goku Heights",
      "city": "Kivzupzi",
      "postcode": "J5C 8H1",
      "__v": 0
    }
  ]
}