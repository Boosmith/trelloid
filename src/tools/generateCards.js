const randomWords = require("random-words");
const bcrypt = require("bcrypt");
const { users } = require("./mockData");

const words = randomWords({ exactly: 10, wordsPerString: 2 });

var mongoObjectId = function() {
	var timestamp = ((new Date().getTime() / 1000) | 0).toString(16);
	return (
		timestamp +
		"xxxxxxxxxxxxxxxx"
			.replace(/[x]/g, function() {
				return ((Math.random() * 16) | 0).toString(16);
			})
			.toLowerCase()
	);
};

const generateCards = function() {
	const cards = [];
	for (let i = 0; i < 10; i++) {
		let card = {
			_id: mongoObjectId(),
			title: words[i],
			boardId: "",
			status: "",
			owner: ""
		};
		cards.push(card);
	}
	console.log(cards);
};

const updateMockUsers = function() {
	const updatedUsers = users.map(user => {
		let userName = randomWords({ exactly: 2, join: " " });
		let password = bcrypt.hashSync(userName, 5);
		let newFields = {
			userName: userName,
			password: password
		};
		return Object.assign(user, newFields);
	});
	console.log(updatedUsers);
};

module.exports =
	// generateCards,
	updateMockUsers();
