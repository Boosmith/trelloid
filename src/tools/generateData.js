const randomWords = require("random-words");
const LoremIpsum = require("lorem-ipsum").LoremIpsum;
const bcrypt = require("bcrypt");
const { users } = require("./mockData");

const words = randomWords({ exactly: 10, wordsPerString: 2 });

var mongoObjectId = function() {
	var timestamp = ((new Date().getTime() / 1000) | 0).toString(16);
	return (
		timestamp +
		"xxxxxxxxxxxxxxxx"
			.replace(/[x]/g, function() {
				return ((Math.random() * 16) | 0).toString(16);
			})
			.toLowerCase()
	);
};

const createLoremIpsum = () => {
	const lorem = new LoremIpsum({
		wordsPerSentence: {
			max: 10,
			min: 4
		}
	});

	return lorem.generateSentences(5);
};

const generateCards = function() {
	const cards = [];
	for (let i = 0; i < 10; i++) {
		let card = {
			_id: mongoObjectId(),
			number: `#${i}`,
			title: words[i],
			description: createLoremIpsum(),
			status: "",
			board: "",
			comments: "",
			activity: "",
			attachments: "",
			owner: {
				_id: "5d42e5d5e1e3d731df26010a",
				firstName: "Andrew",
				lastName: "Smith"
			},
			members: JSON.stringify([
				{
					_id: "5d42104a7979bd071c129606",
					firstName: "Dale",
					lastName: "Fitzgerald"
				},
				{
					_id: "5d42104a7979bd071c129609",
					firstName: "Adeline",
					lastName: "Fitzgerald"
				}
			]),
			createdDate: new Date().toJSON(),
			modifiedDate: new Date().toJSON()
		};
		cards.push(card);
	}
	console.log("object: %O", cards);
};

const generateStatuses = function() {
	const statuses = [];
	for (let i = 0; i < 5; i++) {
		let status = {
			_id: mongoObjectId(),
			status: ""
		};
		statuses.push(status);
	}
	console.log(statuses);
};
const updateMockUsers = function() {
	const updatedUsers = users.map(user => {
		let userName = randomWords({ exactly: 2, join: " " });
		let password = bcrypt.hashSync(userName, 5);
		let newFields = {
			userName: userName,
			password: password
		};
		return Object.assign(user, newFields);
	});
	console.log(updatedUsers);
};

module.exports = generateStatuses();
//updateMockUsers();
