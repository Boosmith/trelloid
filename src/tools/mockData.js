// Cards.
const cards = [
	{
		_id: "5d6aa2c08040a256ce40c16b",
		number: "#0",
		title: "took division",
		description:
			"Veniam amet sint aliqua sit. Id nulla aliquip est ullamco excepteur sint aliquip. Ut in consequat esse incididunt. Laboris consequat pariatur aute incididunt labore elit. Velit veniam aute cupidatat laborum consectetur amet fugiat.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.779Z",
		modifiedDate: "2019-08-31T16:39:28.779Z"
	},
	{
		_id: "5d6aa2c0340c4e6be56eab91",
		number: "#1",
		title: "four perfectly",
		description:
			"Exercitation aute eiusmod culpa officia. Dolor aliquip minim veniam veniam aute. Eiusmod velit magna labore. Consequat quis quis in adipisicing et est ea. Nisi consectetur occaecat in.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.779Z",
		modifiedDate: "2019-08-31T16:39:28.779Z"
	},
	{
		_id: "5d6aa2c0c84791e741e771d8",
		number: "#2",
		title: "eaten chosen",
		description:
			"Consequat minim ipsum eiusmod qui velit proident. Adipisicing sint ad veniam qui reprehenderit consectetur elit dolor aliquip. Ea aliquip in ullamco culpa nisi adipisicing sit. Dolor eiusmod proident aute. Id elit velit aute commodo non ipsum.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.779Z",
		modifiedDate: "2019-08-31T16:39:28.779Z"
	},
	{
		_id: "5d6aa2c0f16942ff1f889661",
		number: "#3",
		title: "plant build",
		description:
			"Ex voluptate irure proident non commodo. Est eiusmod magna magna qui non magna. Officia duis et non amet dolore. Quis culpa ex consectetur magna cupidatat. Eu do labore tempor.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.779Z",
		modifiedDate: "2019-08-31T16:39:28.779Z"
	},
	{
		_id: "5d6aa2c0a8d87097a70116ba",
		number: "#4",
		title: "house mirror",
		description:
			"Consequat ullamco reprehenderit non eiusmod. Reprehenderit quis commodo commodo pariatur proident. Dolore fugiat ullamco dolor minim ea excepteur. Irure qui velit deserunt dolore sunt duis. Non officia aliquip nulla adipisicing excepteur eu quis quis.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	},
	{
		_id: "5d6aa2c00f0395222d6483ba",
		number: "#5",
		title: "new dawn",
		description:
			"Deserunt consequat nisi sunt et sint eu magna. Pariatur pariatur ex aliqua dolor Lorem fugiat nulla et veniam. Sunt nostrud est duis reprehenderit culpa esse occaecat pariatur. Aliquip nostrud occaecat officia laboris aute et cillum non ex. Ea ad sunt non adipisicing laborum fugiat sint.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	},
	{
		_id: "5d6aa2c07102a470b4f1066c",
		number: "#6",
		title: "income whose",
		description:
			"Ex officia est pariatur officia irure eu pariatur mollit. Sint excepteur cupidatat sunt do sit consequat non. Aute ea aliqua id do ut in. Non reprehenderit qui ea labore cupidatat excepteur excepteur qui aliquip. Mollit id labore voluptate nisi aute.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	},
	{
		_id: "5d6aa2c09cad9ad87fb52743",
		number: "#7",
		title: "occur compare",
		description:
			"Mollit ut sunt sit. Commodo commodo sit esse anim amet. Culpa nostrud exercitation cupidatat. Commodo duis esse aliqua non. Elit laborum consequat quis ullamco nulla in aliqua est.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	},
	{
		_id: "5d6aa2c0b3c3b66d9e80ceb6",
		number: "#8",
		title: "hold how",
		description:
			"Id fugiat et aliqua nisi. Ea laborum dolor exercitation sit. Veniam ea dolore ad est ad anim excepteur. Eiusmod magna aliquip duis. Anim deserunt dolor eu velit eiusmod aliquip eu consectetur Lorem.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	},
	{
		_id: "5d6aa2c07500c1f9ceb14f9b",
		number: "#9",
		title: "upon fourth",
		description:
			"Fugiat nulla tempor pariatur anim fugiat culpa est ut cillum. Excepteur commodo reprehenderit mollit dolor fugiat reprehenderit. Magna culpa enim esse Lorem occaecat esse culpa ullamco. Tempor culpa voluptate fugiat reprehenderit. Ex non laborum ullamco sit nostrud laborum Lorem.",
		status: "",
		board: "",
		comments: "",
		activity: "",
		attachments: "",
		owner: {
			_id: "5d42e5d5e1e3d731df26010a",
			firstName: "Andrew",
			lastName: "Smith"
		},
		members: [
			{
				_id: "5d42104a7979bd071c129606",
				firstName: "Dale",
				lastName: "Fitzgerald"
			},
			{
				_id: "5d42104a7979bd071c129609",
				firstName: "Adeline",
				lastName: "Fitzgerald"
			}
		],
		createdDate: "2019-08-31T16:39:28.780Z",
		modifiedDate: "2019-08-31T16:39:28.780Z"
	}
];

const newCard = {
	number: "",
	title: "",
	description: "",
	status: "",
	board: "",
	comments: "",
	activity: "",
	attachments: "",
	owner: "",
	members: [[Object], [Object]],
	createdDate: "2019-08-31T16:21:20.029Z",
	modifiedDate: "2019-08-31T16:21:20.029Z"
};

// Users.
const users = [
	{
		_id: "5d42e5d5e1e3d731df26010a",
		firstName: "Andrew",
		lastName: "Smith",
		address: "32 Walpole Road",
		city: "London",
		postcode: "N17",
		__v: 0,
		userName: "claws circle",
		password: "$2b$05$2.3R3gIiLY0.k3W2NyFU1eS0By2gEatUxvFNZ/hm6w.uLP4xynrZK"
	},
	{
		_id: "5d43fed51d0825849c0f0b67",
		firstName: "Rosa",
		lastName: "Smith",
		address: "32 Walpole Road",
		city: "London",
		postcode: "N17",
		__v: 0,
		userName: "court nation",
		password: "$2b$05$N8m6IdK07d.ebxqkS2pvDuBAh8oVkeVblDr/D36u5GdK3eKa9OrWy"
	},
	{
		_id: "5d42104a7979bd071c129604",
		firstName: "Dora",
		lastName: "Fitzgerald",
		address: "1300 Eknaw Path",
		city: "Evegirhe",
		postcode: "P8K 4T1",
		__v: 0,
		userName: "brass indeed",
		password: "$2b$05$CB4O9hzF7Wx.C..hActQoOzjIiFVcKK79h6zJutK1seHj13b9i/gq"
	},
	{
		_id: "5d42104a7979bd071c129605",
		firstName: "Bryan",
		lastName: "Fitzgerald",
		address: "1050 Dedu Point",
		city: "Irhekde",
		postcode: "G1K 2N9",
		__v: 0,
		userName: "interior one",
		password: "$2b$05$u3S3FSfVetkUrYAbQgUTROLbk7l4Ez3Z6PcQlp1XRiPoy83ult.BW"
	},
	{
		_id: "5d42104a7979bd071c129606",
		firstName: "Dale",
		lastName: "Fitzgerald",
		address: "560 Fibab Glen",
		city: "Hudmakap",
		postcode: "T9U 0N6",
		__v: 0,
		userName: "season carried",
		password: "$2b$05$d1NOjWSsLqvjssyI1LKaK.QcnofuAwjgIozR86Pn/p2ixA0CjVX06"
	},
	{
		_id: "5d42104a7979bd071c129607",
		firstName: "Alma",
		lastName: "Fitzgerald",
		address: "1630 Zuas Place",
		city: "Caturaagu",
		postcode: "G9E 7N0",
		__v: 0,
		userName: "stay underline",
		password: "$2b$05$MqkYhqxPam.a2LuUB7J9l.nDih2DRRIYxGoxh4SLzKxvA7W7Ltmjm"
	},
	{
		_id: "5d42104a7979bd071c129608",
		firstName: "Billy",
		lastName: "Fitzgerald",
		address: "1689 Hopge Square",
		city: "Vinpati",
		postcode: "C2O 6D1",
		__v: 0,
		userName: "end owner",
		password: "$2b$05$1ErQO/LyTVZzPqXy5fv2AOEQUBcw0cypVQffxXX7J/1qJphFckDbO"
	},
	{
		_id: "5d42104a7979bd071c129609",
		firstName: "Adeline",
		lastName: "Fitzgerald",
		address: "1912 Dusved Ridge",
		city: "Atehasma",
		postcode: "N2D 7C3",
		__v: 0,
		userName: "column location",
		password: "$2b$05$BHvS9h08epkH8LL7CJ31HuCz09fYJ3DIv81w0BWZQu.mtoB0v7BV2"
	},
	{
		_id: "5d42104a7979bd071c12960a",
		firstName: "Charlotte",
		lastName: "Fitzgerald",
		address: "611 Gumaf Way",
		city: "Kalover",
		postcode: "E4Y 2E7",
		__v: 0,
		userName: "action please",
		password: "$2b$05$8TrKFk/gSO241uG/4s6k.uJWi/0tc0KZPCoqbVryAfVu/7YfgDqvy"
	},
	{
		_id: "5d42104a7979bd071c12960b",
		firstName: "Rodney",
		lastName: "Fitzgerald",
		address: "1961 Goku Heights",
		city: "Kivzupzi",
		postcode: "J5C 8H1",
		__v: 0,
		userName: "solution primitive",
		password: "$2b$05$ZQXo4wwzA.wzrGKY7/BM8uthaaoYXvkFx8HbbUDlPTwXMAGZOKsu."
	}
];
const newUser = {
	firstName: "",
	lastName: "",
	address: "",
	city: "",
	postcode: "",
	userName: "",
	password: ""
};

module.exports = {
	cards,
	newCard,
	newUser,
	users
};
