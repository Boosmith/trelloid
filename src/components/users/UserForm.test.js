import React from "react";
import UserForm from "./UserForm";
import renderer from "react-test-renderer";
import { users } from "../../../tools/mockData";
import { isTSAnyKeyword, exportAllDeclaration } from "@babel/types";
import { JestEnvironment } from "@jest/environment";

it("sets submit button to 'Saving...' when saving is true", () => {
	const tree = renderer.create(
		<UserForm user={users[0]} onSave={jest.fn()} onChange={jest.fn()} saving />
	);

	expect(tree).toMatchSnapshot();
});
