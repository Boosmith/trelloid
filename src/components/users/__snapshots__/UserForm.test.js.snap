// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button to 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     User
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="firstName"
    >
      First name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="firstName"
        onChange={[MockFunction]}
        type="text"
        value="Andrew"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="lastName"
    >
      Last name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="lastName"
        onChange={[MockFunction]}
        type="text"
        value="Smith"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="address"
    >
      Address
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="address"
        onChange={[MockFunction]}
        type="text"
        value="32 Walpole Road"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="city"
    >
      City
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="city"
        onChange={[MockFunction]}
        type="text"
        value="London"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="postcode"
    >
      Postcode
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="postcode"
        onChange={[MockFunction]}
        type="text"
        value="N17"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
